src/dials/algorithms/scaling/scaling_library.py

```diff
@@ -372,8 +374,103 @@ def determine_best_unit_cell(experiments):
     return best_unit_cell
 
 
+class ExtendedDatasetStatistics(iotbx.merging_statistics.dataset_statistics):
+
+    """A class to extend iotbx merging statistics."""
+
+    def __init__(self, *args, additional_stats=False, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.r_split = None
+        self.r_split_binned = None
+        self.binner = None
+        if not additional_stats:
+            return
+        i_obs = kwargs.get("i_obs")
+        n_bins = kwargs.get("n_bins", 20)
+        if not i_obs:
+            return
+        i_obs_copy = i_obs.customized_copy()
+        i_obs_copy.setup_binner(n_bins=n_bins)
+        i_obs = i_obs.map_to_asu()
+        i_obs = i_obs.sort("packed_indices")
+
+        seed = 0
+
+        split_datasets = split_unmerged(
+            unmerged_indices=i_obs.indices(),
+            unmerged_data=i_obs.data(),
+            unmerged_sigmas=i_obs.sigmas(),
+            seed=seed,
+        )
+        indices = split_datasets.indices
+        m1 = miller.array(
+            miller_set=miller.set(i_obs.crystal_symmetry(), indices),
+            data=split_datasets.data_1,
+        )
+        m2 = miller.array(
+            miller_set=miller.set(i_obs.crystal_symmetry(), indices),
+            data=split_datasets.data_2,
+        )
+        assert i_obs_copy.binner() is not None
+        self.binner = i_obs_copy.binner()
+        m1.use_binning(self.binner)
+        m2.use_binning(self.binner)
+
+        self.r_split = self.calc_rsplit(
+            m1, m2, assume_index_matching=True, use_binning=False
+        )
+        self.r_split_binned = self.calc_rsplit(
+            m1, m2, assume_index_matching=True, use_binning=True
+        )
+
+    def as_dict(self):
+        d = super().as_dict()
+        if not self.r_split:
+            return d
+        d["overall"]["r_split"] = self.r_split
+        d["r_split"] = self.r_split_binned
+        return d
+
+    @classmethod
+    def calc_rsplit(cls, this, other, assume_index_matching=False, use_binning=False):
+        # based on White, T. A. et al. J. Appl. Cryst. 45, 335-341 (2012).
+        # adapted from cctbx_project/xfel/cxi_cc.py
+        if not use_binning:
+            assert other.indices().size() == this.indices().size()
+            if this.data().size() == 0:
+                return 0.0
+
+            if assume_index_matching:
+                (o, c) = (this, other)
+            else:
+                (o, c) = this.common_sets(other=other, assert_no_singles=True)
+
+            den = flex.sum(o.data() + c.data())
```

Comment:
Should the denominator have an extra factor 0.5 in? From equation (2) in [White et al. 2012](https://journals.iucr.org/j/issues/2012/02/00/db5097/index.html):
![Screenshot 2023-01-17 at 16 05 21](https://user-images.githubusercontent.com/2810624/212950308-c29b0e3f-b013-4603-878d-9bd357754507.png)
Perhaps I'm missing something, but I don't see where the factor 0.5 is in your implementation...

Reply:
This is confusing but I believe I'm correct: 2^(-0.5) / (1/2) = (1 / sqrt2) * 2 = sqrt2. So I just put 2^(+0.5) in the numerator and therefore don't need any factor in the denominator.